src/containers/create-new/index.tsx

```diff
@@ -62,22 +67,28 @@ const CreateNewArea = ({ className, space }: Props) => {
     const submitBtn = target.localName === 'span' ? target.parentElement : target;
     const isPreviewBtn = submitBtn.dataset?.btn;
     setHasImageError(!selectedImage);
+    if (tags.length > 8) setTagErrMessage(tagsErrorMessage.CantMore);
+    if (tags.length === 0) setTagErrMessage(tagsErrorMessage.AtLeast);
     if (isPreviewBtn && selectedImage) {
       setPreviewData({ ...data, blob: selectedImage });
       setShowProductModal(true);
     }
-    if (!isPreviewBtn && selectedImage) {
+    if (!isPreviewBtn && selectedImage && tags.length > 0 && tags.length <= 8) {
       dispatch(mintNft({ ...data, address, blob: selectedImage } as ItemRequest));
       resetForm();
     }
   };
 
   const handleTagsChange = (tagProps: string[]) => {
     setTags(tagProps);
-    if (tags.length + 1 === 8) setTagErrMessage('You can not set more than 8 tags');
-    else setTagErrMessage('');
   };
 
+  useMemo(() => {
```

Comment:
useMemo is meant to return a memoized value. In this case all you are doing is a validation and setting a state variable. I would suggest to change this for a useEffect instead.